app/controllers/api/connect/v4/systems/products_controller.rb

```diff
@@ -1,2 +1,24 @@
 class Api::Connect::V4::Systems::ProductsController < Api::Connect::V3::Systems::ProductsController
+
+  def destroy
+    if @product.product_type == 'base'
+      untranslated = N_('The product "%s" is a base product and cannot be deactivated')
+      respond_with_error message: (untranslated % @product.name), localized_message: (_(untranslated) % @product.name)
```

Comment:
I understand that this `respond_with_error` stuff is copy-pasta, but let's not copy-paste bad code 😛 
Rails already provides a [mechanism for rescuing from exceptions -- `rescue_from`](http://api.rubyonrails.org/classes/ActiveSupport/Rescuable/ClassMethods.html#method-i-rescue_from). We can simply raise a `TranslatedError` exception wherever we face an error and be done with it. We won't need those `and return` which are littered around the code, and we'll just have one standard way of reacting to errors.

I propose we implement something like [this](https://github.com/SUSE/potato/compare/deregistration...rescue-from?expand=1) (you can just cherry-pick this commit if you like)